app/models/user.rb

```diff
@@ -1,5 +1,6 @@
 class User < ActiveRecord::Base
   authenticates_with_sorcery!
+  before_create :create_organiser_token
```

Comment:
1. yes if user is successfully updates please set this token to nil
2. you need to change this query instead of `@organisers = User.where(role: 1)` you need to use `@organisers = User.where(role: 1, organiser_token: nil)` for confirmed organisers and `@organisers = User.where(role: 1).where.not(organiser_token: nil)` for pending ones